jmh-core/src/main/java/org/openjdk/jmh/runner/Runner.java

```diff
@@ -390,15 +390,18 @@ private BenchmarkParams newBenchmarkParams(BenchmarkListEntry benchmark, ActionM
                 benchmark.getThreads().orElse(
                         Defaults.THREADS));
 
-        if (threads == Threads.MAX) {
+        if (threads == Threads.MAX || threads == Threads.HALF_MAX) {
             if (cpuCount == 0) {
                 out.print("# Detecting actual CPU count: ");
                 cpuCount = Utils.figureOutHotCPUs();
                 out.println(cpuCount + " detected");
             }
-            threads = cpuCount;
+            if (threads == Threads.HALF_MAX) {
+                threads = Math.max(1, cpuCount / 2);
```

Comment:
I'd say we just round half-up: `(cpuCount + 1) / 2`, which makes a better choice for odd number of CPUs, including `1`.